### PR TITLE
(feat): Adding openebs pool network loss experiment

### PR DESCRIPTION
**Pull request by rajdas98**
**What this PR does / why we need it**:
- Adding openEBS pool network loss experiment
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Checklist**

* [x] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+- name: Pick a cStor target pod belonging to the PV
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml]: For pool-network-loss, Which one is required pool-pod or target-pod?

File: experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  register: cstor_target_pod
+
+# including pumba lib  -> network_chaos
+- name: Inject egress network loss on cstor-pool container for {{ chaos_duration }}ms
+  include_tasks: /chaoslib/pumba/network_chaos/network_chaos.yml
+  vars:
+    n_interface: "eth0"
+    n_packet_loss: "{{ packet_loss_perc }}"
+    c_container: "cstor-pool"
+    app_pod: "{{ cstor_target_pod.stdout }}"
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml]: same as above

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+  vars_files:
+    - /mnt/parameters.yml
+    - /experiments/openebs/openebs_components.yml
+  
+  tasks:
+    - block:
+      - include_tasks: /utils/runtime/create_testname.yml  
+
+      - name: Identify the chaos util to be invoked
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml]: Can you please add this step in openebs_pool_network_loss_ansible_prerequisites.yml file?

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      - include_tasks: /utils/runtime/create_testname.yml  
+
+      - name: Identify the chaos util to be invoked 
+        template:
+          src: chaosutil.j2
+          dest: chaosutil.yml
+
+      - name: Identify the data consistency util to be invoked
+        template:
+          src: data_persistence.j2
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml]: Can you please add this step in openebs_pool_network_loss_ansible_prerequisites.yml file?

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+      # DISPLAY APP INFORMATION
+      - name: Display the app information passed via the test job
+        debug:
+          msg:
+            - "The application info is as follows:"
+            - "Namespace    : {{ a_ns }}"
+            - "Label        : {{ a_label }}"
+            - "PVC          : {{ a_pvc }}"
+            # - "StorageClass : {{ sc }}"
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml]: remove this comment

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+            
+      # Pre req
+      - block:
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml]: Can you please add this step in openebs_pool_network_loss_ansible_prerequisites.yml file?

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+        - name: Verify that the AUT (Application Under Test) is running
+          include_tasks: "/utils/common/status_app_pod.yml"
+          vars:
+            app_ns: "{{ a_ns }}"
+            app_label: "{{ a_label }}"      
+            delay: 5
+            retries: 60
+
+        - name: Get application pod name
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml]: How this task is different from the task present on `line 110`

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+        - name: Create some test data
+          include: "{{ consistencyutil }}"
+          vars:
+            status: 'LOAD'
+            ns: "{{ a_ns }}"
+            pod_name: "{{ app_pod.stdout }}"
+          when: data_persistence | length > 0
+
+        - set_fact:
+            flag: "Pass"
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml]: please use same caps `pass`

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+          
+      rescue:
+        - set_fact:
+            flag: "Fail"
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_ansible_logic.yml]: fail

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 10]
```diff
@@ -0,0 +1,79 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: openebs-pool-network-loss
+data:
+  parameters.yml: |
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: please don't provide any data inside this

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+metadata:
+  generateName: openebs-pool-network-loss- 
+  namespace: litmus
+spec:
+  template:
+    metadata:
+      labels:
+        name: openebs-target-network-loss
+    spec:
+      serviceAccountName: litmus
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: %CHAOS_SERVICE_ACCOUNT%

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+      restartPolicy: Never
+      containers:
+      - name: ansibletest
+        image: imrajdas/ansible-runner:test
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: `litmuschaos/ansible-runner:ci`

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+          - name: ANSIBLE_STDOUT_CALLBACK
+            value: 'default'
+
+          # provide openebs namespace
+          - name: OPENEBS_NAMESPACE
+            value: 'openebs'
+ 
+          # # provide application namespace
+          - name: APP_NAMESPACE
+            value: 'default'
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: Please remove the values from Env

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+
+           # provide application label
+          - name: APP_LABEL
+            value: 'app=busybox'
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: same as above

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+           # provide application pvc
+          - name: APP_PVC
+            value: 'demo-vol1-claim'
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: same as above

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+            value: '240000' # in milliseconds
+
+          - name: LIVENESS_APP_LABEL
+            value: ''
+
+          - name: LIVENESS_APP_NAMESPACE
+            value: ''
+
+          - name: DATA_PERSISTENCE
+            value: 'busybox'
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: same as above

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+           # only pumba supported
+           # For pumba image use : gaiaadm/pumba:0.6.5
+          - name: LIB_IMAGE  
+            value: 'gaiaadm/pumba:0.6.5' 
+
+          - name: NETWORK_PACKET_LOSS_PERCENTAGE
+            value: '100' # in percentage
+
+          - name: TOTAL_CHAOS_DURATION
+            value: '240000' # in milliseconds
```
Comment (ksatchit) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: 120000

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 18]
```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: openebs-pool-network-loss
+data:
+# Refer docs for this data.
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: pls mention parameters.yml file name here

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+  template:
+    metadata:
+      labels:
+        name: openebs-pool-network-loss
+    spec:
+      serviceAccountName: %CHAOS_SERVICE_ACCOUNT%
+      restartPolicy: Never
+      containers:
+      - name: ansibletest
+        image: litmuschaos/ansible-runner:test
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: litmuschaos/ansible-runner:ci

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+        imagePullPolicy: Always
+        env: 
+          - name: ANSIBLE_STDOUT_CALLBACK
+            value: 'default'
+
+          # provide openebs namespace
+          - name: OPENEBS_NAMESPACE
+            value: ''
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: provide default value here `openebs`

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 21]
```diff
@@ -0,0 +1,78 @@
...
+            value: 'gaiaadm/pumba:0.6.5' 
+
+          - name: NETWORK_PACKET_LOSS_PERCENTAGE
+            value: '100' # in percentage
+
+          - name: TOTAL_CHAOS_DURATION
+            value: '120000' # in milliseconds
+          
+          - name: CHAOS_INTERVAL
+            value: '100'
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: pls check this value

File: experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+  - name: Display pool pod name
+    debug: 
+      msg: 
+        - "The application info is as follows:"
+        - " pod-name  : {{ cstor_pool_pod }}"
+
+  - name: Inject egress network loss on cstor-pool container for {{ chaos_duration }}ms
+    include_tasks: /chaoslib/pumba/network_chaos/network_chaos.yml
+    vars:
+      n_interface: "eth0"
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml]:
```suggestion
      n_interface: "{{ stg_interface }}"
```

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+
+          - name: LIVENESS_APP_LABEL
+            value: ''
+
+          - name: LIVENESS_APP_NAMESPACE
+            value: ''
+
+          - name: DATA_PERSISTENCE
+            value: 'busybox'
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]:
```suggestion
            value: ''
```

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+      - name: ansibletest
+        image: litmuschaos/ansible-runner:test
+        imagePullPolicy: Always
+        env: 
+          - name: ANSIBLE_STDOUT_CALLBACK
+            value: 'default'
+
+          # provide openebs namespace
+          - name: OPENEBS_NAMESPACE
+            value: ''
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]:
```suggestion
            value: 'openebs'
```

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+  template:
+    metadata:
+      labels:
+        name: openebs-pool-network-loss
+    spec:
+      serviceAccountName: %CHAOS_SERVICE_ACCOUNT%
+      restartPolicy: Never
+      containers:
+      - name: ansibletest
+        image: litmuschaos/ansible-runner:test
```
Comment (shubhamchaudhary) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]:
```suggestion
        image: litmuschaos/ansible-runner:ci
```

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml (hunk 23 above)
Comment (rajdas98) [reply experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: done

File: experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml (hunk 22 above)
Comment (rajdas98) [reply experiments/openebs/openebs-pool-network-loss/cstor_pool_network_loss.yml]: We can have this in next release

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml (hunk 24 above)
Comment (rajdas98) [reply experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: done

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+          - name: APP_PVC
+            value: ''
+
+           # only pumba supported
+           # For pumba image use : gaiaadm/pumba:0.6.5
+          - name: LIB_IMAGE  
+            value: 'gaiaadm/pumba:0.6.5' 
+
+          - name: NETWORK_PACKET_LOSS_PERCENTAGE
+            value: '10000' # in percentage
```
Comment (chandankumar4) [inline experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: Can you check the value!!

File: experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml (hunk 24 above)
Comment (chandankumar4) [reply experiments/openebs/openebs-pool-network-loss/openebs_pool_network_loss_k8s_job.yml]: Make the `openebs` as default if user want to override they will pass the env

**Pull request merged**